canonical-web-and-design/maas-ui
# Update images page to allow selecting architectures on a per-release basis.

##  Done
- Updated images page to allow selecting architectures on a per-release basis. Originally, selecting releases and architectures would download every possible combination.
- Disable release/arch combinations that are unsupported.
- Rearranged markup in boot_images.html to make a bit more sense.

## QA
- Run a local MAAS using [this branch](https://code.launchpad.net/~ltrager/maas/+git/maas/+merge/385331), which updates the websocket handler to allow for per-release selections.
- Go to /MAAS/l/images and check that you can select architectures per release, and the table below displays the summary correctly.
- Check that 20.04/i386 is disabled with a tooltip.
- Click "Save selection" and check that the downloads start correctly.
- Check that once the downloads are complete you can also remove them correctly.

## Fixes
Fixes #1159
Fixes canonical-web-and-design/maas-design#910

## Launchpad Issue
[lp#1877585](https://bugs.launchpad.net/maas/+bug/1877585)

## Screenshot
![0 0 0 0_8400_MAAS_l_intro_user](https://user-images.githubusercontent.com/25733845/84233733-3ee4ff00-ab36-11ea-86a9-09f9c9876cd1.png)
